dependencies/influxdb/http.go

```diff
@@ -441,3 +458,64 @@ func (h seriesCardinalityHttpReader) Read(ctx context.Context, f func(flux.Table
 	}
 	return h.Query(ctx, f, &file, h.Bounds.Now, mem)
 }
+
+type httpWriter struct {
+	writer      *api.WriteAPIImpl
+	errChan     <-chan error
+	latestError chan error
+}
+
+func newHttpWriter(writer *api.WriteAPIImpl) (*httpWriter, error) {
+	w := &httpWriter{
+		writer:      writer,
+		errChan:     writer.Errors(),
+		latestError: make(chan error, 1),
+	}
+	go func() {
+		for err := range w.errChan {
+			if err != nil {
+				select {
+				case w.latestError <- err:
+				default:
+				}
+			}
+		}
+		close(w.latestError)
+	}()
+	return w, nil
+}
+
+var _ Writer = &httpWriter{}
+
+// Write sends points asynchronously to the underlying write api.
+// Errors are returned only on a best-effort basis.
+func (h *httpWriter) Write(metric ...protocol.Metric) error {
+	buf := new(bytes.Buffer)
```

Comment:
I think this ends up functionally the same because the escape checker finds out that it's the same but the normal way to deal with this is:

```
var buf bytes.Buffer
enc := protocol.NewEncoder(&buf)
```

It's fine to leave it as-is.

Reply:
In C++/Rust I would expect the two to be different, but in go once you take the address you need the initial allocation on the heap anyway, unless the go escape checker can validate it doesn't escape (e.g. that NewEncoder doesn't store the pointer on the heap somewhere).

Reply:
Noted about the preferred style though, thanks.